stock_picking_mto/models/stock_picking.py

```diff
@@ -0,0 +1,40 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    def _get_selection_procure_method(self):
+        return self.env['stock.move'].fields_get(
+            allfields=['procure_method'])['procure_method']['selection']
+
+    def _compute_procure_method(self):
+        for picking in self:
+            procure_method = False
+            for move in picking.move_lines:
+                if not procure_method:
+                    procure_method = move.procure_method
+                elif procure_method != move.procure_method:
```

Comment:
@chienandalu Why this kind of logic? If you want to get the first move procure method in picking:
```python
for picking in self:
    for move in picking.move_lines:
        picking.procure_method = move.procure_method
        break
```
As procure_method in moves is required, you won't have False values.

Reply:
That's not the intention: this is going to be a tri-valued field: empty if there are different values across moves, or one of the 2 possible values if they are all the same.

Reply:
@pedrobaeza Ok, but the _get_selection_procure_method is missing the empty value.

Moreover, written like that, picking.procure_method will always have the first move value...

I would have written it like this :

```python
# ensure you have at least one move line - I'm not sure a picking could be void
for picking in self.filtered('move_lines'):
    first_move = picking.move_lines[0]
    procure_method = first_move.procure_method
    if any(move.procure_method != procure_method for move in picking.move_lines - first_move):
        procure_method = ''  # Not False, empty is '' in selection
    picking.procure_method = procure_method
```
It's clearer to read and the use of any is more efficent

Reply:
@rousseldenis No, the empty value is False, and it's obtained if no moves, or if there's any discrepancy. The problem is one break that is not correctly indented.

Reply:
Ok understand

Reply:
But my method is correct as if there is no moves, the value is initialized with False.

Reply:
Your method is less efficient and I see it less readable.

Reply:
Ok, that was a little to optimize as we are used to huge pickings, so we make attention to such computations.

But another way (:innocent: ):
```python
for picking in self.filtered('move_lines'):
    procure_methods = picking.mapped('move_lines.procure_method')
    if len(procure_methods) == 1:
        picking.procure_method = procure_methods[0]
```
More readable :smiley:

Reply:
This is not going to work, as these aren't recordsets, and values are not "joined". You can get `['mto', 'mto', 'mto',...]`